### ISSUE TITLE
Pods must be called when the executor dies

### ISSUE THREAD

**Issue by sttts** [mesosphere/kubernetes-mesos#573]

**Comment by karlkfi** [comment]
If the executor dies prematurely does it get resurrected?
Does this assertion apply to "dying" or just when deliberately stopped?
The pods should at least be killed when the executor si told to shut down as part of the uninstallation of k8s.

**Comment by sttts** [comment]
Our minion code restarts the executor/kubelet when it dies. But there are situation (see description) when the minion/executor/kubelet are not restarted.

**Comment by jdef** [comment]
actually, the minion/server.go:launchExecutorServer() code never tries to
restart the executor. once it dies, the minion tears down kube-proxy and
then exits.

**Comment by sttts** [comment]
I thought we had the RC=0 special case. But as we don't support recovering pods after restart, it wouldn't help.

**Comment by jdef** [comment]
xref #410 (duplicate?)

**Comment by jdef** [comment]
google suggested that we implement a garbage collecting process in the slave that, based on docker labels, kills containers that are tagged with an executor id for which no executor exists

**Comment by sttts** [comment]
Good idea. Can you create a Jira for Mesos for this? I guess we won't come up with another solution within k8sm for this. So a mesos-slave extension is probably our only choice.

**Comment by jdef** [comment]
related internal JIRAs:
- KUBERNETES-224 :: I WANT to be assured that docker containers created by my executor(s) will not live indefinitely once my executors are stopped/killed.
- DCOS-4133 :: I WANT to launch docker containers from a custom executor with some guarantee that once my executor is stopped/killed, such docker containers will eventually be garbage collected by DCOS.

**Comment by jdef** [comment]
assigned to Marcus, needs assignment to some internal JIRA epic

**Comment by jdef** [comment]
xref #410
